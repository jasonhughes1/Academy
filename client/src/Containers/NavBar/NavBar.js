import React, { Component } from 'react';
import './NavBar.css';
import { getData } from '../../helpers/API';
import { connect } from 'react-redux';
import { getAllData } from '../../actions/actions';

export class NavBar extends Component {

  componentDidMount() {
    this.props.salaries();
  }

  render() {
    return (
      <div className='nav-bar'>
        <img className="logo" src={require('../../assets/turing.png')} alt="logo" />
        <h1 className='main-header'>Turing Salaries</h1>
      </div>
    );
  }
}

export const mapStateToProps = (store) => {
  return {
    alumni: store.alumData
  };
};

export const mapDispatchToProps = (dispatch) => {
  return {
    salaries: (newData) => {
      dispatch(getAllData(newData));
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(NavBar);
