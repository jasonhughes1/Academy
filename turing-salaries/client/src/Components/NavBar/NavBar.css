.nav-bar {
  background-color: #48a9b6;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.background-image {
  color: transparent;
  background-image: url('../../assets/turing.png');
  background-size: cover;
  height: 175px;
  width: 181px;
}

.background-image:hover {
    -webkit-animation-name: rotate;
    -webkit-animation-duration: 2s;
    -webkit-animation-iteration-count: infinite;
    -webkit-animation-timing-function: linear;
    -moz-animation-name: rotate;
    -moz-animation-duration: 2s;
    -moz-animation-iteration-count: infinite;
    -moz-animation-timing-function: linear;
}

@-webkit-keyframes rotate {
    from {-webkit-transform: rotate(0deg);}
    to {-webkit-transform: rotate(360deg);}
}

.main-header {
  margin-top: -10px;
}
