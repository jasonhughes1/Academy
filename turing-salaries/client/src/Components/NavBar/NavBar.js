import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';

class NavBar extends Component {
  render() {
    
    return (
      <div className='nav-bar'>
    <h1 className='main-header'>Turing Salaries</h1>
    <NavLink className='nav' exact to='/'>Home</NavLink>
    <NavLink className='nav' to='/viewsalary'>ViewSalary</NavLink>
    <NavLink className='nav' to='/createsalary'>CreateSalary</NavLink>
  </div>
    )
  }
}

export default NavBar;
