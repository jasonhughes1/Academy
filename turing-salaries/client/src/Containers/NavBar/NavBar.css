.nav-bar {
  align-items: center;
  background-color: #48a9b6;
  display: flex;
  justify-content: center;
  flex-direction: row;
}

.logo {
  justify-content: center;
  height: 100px;
  margin: 15px 0px;
  width: 100px;
}

.logo:hover {
    -webkit-animation-name: rotate;
    -webkit-animation-duration: 2s;
    -webkit-animation-iteration-count: infinite;
    -webkit-animation-timing-function: linear;
    -moz-animation-name: rotate;
    -moz-animation-duration: 2s;
    -moz-animation-iteration-count: infinite;
    -moz-animation-timing-function: linear;
}

@-webkit-keyframes rotate {
    from {-webkit-transform: rotate(0deg);}
    to {-webkit-transform: rotate(360deg);}
}

.main-header {
  font-size: 46px;
  justify-content: center;
  margin: 0px;
}
