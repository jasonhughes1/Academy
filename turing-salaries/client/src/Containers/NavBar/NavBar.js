import React, { Component } from 'react';
import './NavBar.css';
import API from '../../helpers/API';
const { getData } = API;

class NavBar extends Component {

  async componentDidMount() {
    let alumniData = await getData()
    console.log(alumniData);
  }

  render() {
    return (
      <div className='nav-bar'>
        <img className="logo" src={require('../../assets/turing.png')} alt="logo" />
        <h1 className='main-header'>Turing Salaries</h1>
      </div>
    )
  }
}

export default NavBar;
