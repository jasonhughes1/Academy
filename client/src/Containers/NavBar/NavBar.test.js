import React from 'react';
import { shallow } from 'enzyme';
import { NavBar, mapStateToProps, mapDispatchToProps } from './NavBar';

describe('NavBar Tests', () => {

  it.skip('should be defined', () => {
    const mockData = [];
    const renderedNavBar = shallow(<NavBar alumni={mockData} />);
    expect(renderedNavBar).toBeDefined()
  });

  it.only('should match snap shot', () => {
    const mockData = [];
    const renderedNavBar = shallow(<NavBar alumni={mockData} />);

    expect(renderedNavBar).toMatchSnapshot();
  })
})

describe('Map State To Props Test', () => {
  it.skip('should alumni data from the store', () => {
    const mockStore = {
      alumni: {
      "original_cohort": 1502,
      "original_start_date": "2/9/2015",
      "final_cohort": 1507,
      "grad_date": "1/22/2016",
      "accepted_date": "1/22/2016",
      "days_to_offer": 0,
      "payment_type": "Annual",
      "payment_rate": 100000.00,
      "annualized_salary": 100000.00,
      "title": "Research Associate"
      }
    }
    const expected = mapStateToProps(mockStore);
    expect(expected.alumni).toEqual(mockStore.alumni)
  })
})
