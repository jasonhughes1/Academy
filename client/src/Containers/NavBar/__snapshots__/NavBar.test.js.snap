// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Tests should match snap shot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar
    alumni={Array []}
    getAllData={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="logo"
          className="logo"
          src="turing.png"
/>,
        <h1
          className="main-header"
>
          Turing Salaries
</h1>,
      ],
      "className": "nav-bar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "logo",
          "className": "logo",
          "src": "turing.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Turing Salaries",
          "className": "main-header",
        },
        "ref": null,
        "rendered": "Turing Salaries",
        "type": "h1",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="logo"
            className="logo"
            src="turing.png"
/>,
          <h1
            className="main-header"
>
            Turing Salaries
</h1>,
        ],
        "className": "nav-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "logo",
            "className": "logo",
            "src": "turing.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Turing Salaries",
            "className": "main-header",
          },
          "ref": null,
          "rendered": "Turing Salaries",
          "type": "h1",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
